pkg/pillar/cmd/zedagent/handlenetworkinstance.go

```diff
@@ -696,3 +701,128 @@ func publishInfoToZedCloud(UUID string, infoMsg *zinfo.ZInfoMsg, iteration int)
 		writeSentDeviceInfoProtoMessage(data)
 	}
 }
+
+func handleAppFlowMonitorModify(ctxArg interface{}, key string,
+	statusArg interface{}) {
+
+	log.Infof("handleAppFlowMonitorModify(%s)\n", key)
+	flows := cast.CastFlowStatus(statusArg)
+	if flows.Key() != key {
+		log.Errorf("handleAppFlowMonitorModify key/UUID mismatch %s vs %s; ignored %+v\n",
+			key, flows.Key(), flows)
+		return
+	}
+
+	flowIteration++
+	// encoding the flows with protobuf format
+	pflows := protoEncodeAppFlowMonitorProto(flows)
+
+	// send protobuf to zedcloud
+	sendFlowProtobuf(pflows, flowIteration)
+}
+
+func handleAppFlowMonitorDelete(ctxArg interface{}, key string,
+	statusArg interface{}) {
+
+	log.Infof("handleAppFlowMonitorDelete(%s)\n", key)
+	flows := cast.CastFlowStatus(statusArg)
+	if flows.Key() != key {
+		log.Errorf("handleAppFlowMonitorDelete key/UUID mismatch %s vs %s; ignored %+v\n",
+			key, flows.Key(), flows)
+		return
+	}
+	log.Infof("handleAppFlowMonitorDelete(%s) done\n", key)
+}
+
+func protoEncodeAppFlowMonitorProto(ipflow types.IPFlow) *zflow.FlowMessage {
```

Comment:
What is the maximum size of a flow message? 
Note that ngnix is configured to drop messages larger than 32k or so, and this code doesn't implement any upper limit. logmanager is an example of code with an upper limit, and it sends separate messages until all the information has been sent.

Reply:
how about let me figure out for each entry, approximate size, and limit to that size, but come back later to handle separate messages

Reply:
testing showing approximately 100 bytes per flow, so I'll limit this to 280 flows/DNS entries per upload to the cloud at a time.